### PR TITLE
[MI-2922] Fixed the issue of plugin not running with MySQL

### PR DESCRIPTION
**Pull request by manojmalik20**

### CODE REVIEW

File: server/store/store.go [hunk 1]
```diff
@@ -341,3 +388,15 @@ func (s *SQLStore) getQueryBuilder() sq.StatementBuilderType {
 
 	return builder.RunWith(s.db)
 }
+
+// func (s *SQLStore) getQueryBuilder() sq.StatementBuilderType {
```
Comment (raghavaggarwal2308) [inline server/store/store.go:392]: Remove if not required

File: server/store/store.go [hunk 2]
```diff
@@ -55,33 +57,78 @@ func New(db *sql.DB, driverName string, api plugin.API, enabledTeams func() []st
 	}
 }
 
+func (s *SQLStore) CreateIndexForMySQL(tableName, indexName, columnList string) error {
+	query := `select exists(
+			select distinct index_name from information_schema.statistics 
+			where table_schema = DATABASE()
+			and table_name = 'tableName' and index_name = 'indexName'
+		)`
```
Comment (raghavaggarwal2308) [inline server/store/store.go:65]:
```suggestion
	query := `SELECT exists(
			SELECT DISTINCT index_name FROM information_schema.statistics 
			WHERE table_schema = DATABASE()
			AND table_name = 'tableName' AND index_name = 'indexName'
		)`
```
For consistency

**Review by raghavaggarwal2308 — APPROVED** [review]

File: server/store/store.go [hunk 3]
```diff
@@ -55,33 +57,78 @@ func New(db *sql.DB, driverName string, api plugin.API, enabledTeams func() []st
...
+		_, err = s.db.Exec(indexQuery)
+	}
+
+	return err
+}
+
 func (s *SQLStore) Init() error {
-	_, err := s.db.Exec("CREATE TABLE IF NOT EXISTS msteamssync_links (mmChannelID VARCHAR PRIMARY KEY, mmTeamID VARCHAR, msTeamsChannelID VARCHAR, msTeamsTeamID VARCHAR)")
+	var err error
+	_, err = s.db.Exec("CREATE TABLE IF NOT EXISTS msteamssync_links (mmChannelID VARCHAR(255) PRIMARY KEY, mmTeamID VARCHAR(255), msTeamsChannelID VARCHAR(255), msTeamsTeamID VARCHAR(255))")
```
Comment (Kshitij-Katiyar) [inline server/store/store.go:94]:
```suggestion
	if _, err = s.db.Exec("CREATE TABLE IF NOT EXISTS msteamssync_links (mmChannelID VARCHAR(255) PRIMARY KEY, mmTeamID VARCHAR(255), msTeamsChannelID VARCHAR(255), msTeamsTeamID VARCHAR(255))"); err !=nil {
```

File: server/store/store.go [hunk 4]
```diff
@@ -55,33 +57,78 @@ func New(db *sql.DB, driverName string, api plugin.API, enabledTeams func() []st
...
+	_, err = s.db.Exec("CREATE TABLE IF NOT EXISTS msteamssync_users (mmUserID VARCHAR(255) PRIMARY KEY, msTeamsUserID VARCHAR(255), token TEXT)")
 	if err != nil {
 		return err
 	}
 
-	_, err = s.db.Exec("CREATE INDEX IF NOT EXISTS idx_msteamssync_users_msteamsuserid ON msteamssync_users (msTeamsUserID)")
+	if s.driverName == model.DatabaseDriverPostgres {
+		_, err = s.db.Exec("CREATE INDEX IF NOT EXISTS idx_msteamssync_users_msteamsuserid ON msteamssync_users (msTeamsUserID)")
+	} else {
+		err = s.CreateIndexForMySQL("msteamssync_users", "idx_msteamssync_users_msteamsuserid", "msTeamsUserID")
```
Comment (Kshitij-Katiyar) [inline server/store/store.go:116]: Add a TODO comment that in future we may need to do optimise this

File: server/store/store.go [hunk 5]
```diff
@@ -55,33 +57,78 @@ func New(db *sql.DB, driverName string, api plugin.API, enabledTeams func() []st
...
+	}
 	if err != nil {
 		return err
 	}
 
-	_, err = s.db.Exec("CREATE TABLE IF NOT EXISTS msteamssync_posts (mmPostID VARCHAR PRIMARY KEY, msTeamsPostID VARCHAR, msTeamsChannelID VARCHAR, msTeamsLastUpdateAt BIGINT)")
+	_, err = s.db.Exec("CREATE TABLE IF NOT EXISTS msteamssync_posts (mmPostID VARCHAR(255) PRIMARY KEY, msTeamsPostID VARCHAR(255), msTeamsChannelID VARCHAR(255), msTeamsLastUpdateAt BIGINT)")
```
Comment (Kshitij-Katiyar) [inline server/store/store.go:122]: same

**Review by Kshitij-Katiyar — APPROVED** [review]

**Pull request merged**